clients/hadoopfs/src/test/java/io/lakefs/auth/GetCallerIdentityV4PresignerTest.java

```diff
@@ -0,0 +1,66 @@
+package io.lakefs.auth;
+
+import com.amazonaws.Request;
+import com.amazonaws.auth.AWSSessionCredentials;
+import io.lakefs.Constants;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public class GetCallerIdentityV4PresignerTest {
+    public static AWSSessionCredentials newMockAWSCreds() {
+        return new AWSSessionCredentials() {
+            @Override
+            public String getSessionToken() {
+                return "sessionToken";
+            }
+
+            @Override
+            public String getAWSAccessKeyId() {
+                return "accessKeyId";
+            }
+
+            @Override
+            public String getAWSSecretKey() {
+                return "secretKey";
+            }
+        };
+    }
+
+    @Test
+    public void testPresignAsURL() throws Exception {
+        AWSSessionCredentials awsCreds = GetCallerIdentityV4PresignerTest.newMockAWSCreds();
+        GetCallerIdentityV4Presigner stsPresigner = new GetCallerIdentityV4Presigner();
+        GeneratePresignGetCallerIdentityRequest stsReq = new GeneratePresignGetCallerIdentityRequest(
+                new URI("https://sts.amazonaws.com"),
+                awsCreds,
+                new HashMap<String,String>(){{
```

Comment:
Can we use e.g. [ImmutableMap](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableMap.html)?

Reply:
Dependency that exists only in tests we discussed and decided not to introduce into lakeFSFS.